pkg/server/config.go

```diff
@@ -0,0 +1,304 @@
+package server
+
+import (
+	log "github.com/sirupsen/logrus"
+	"golang.org/x/net/context"
+	"github.com/golang/protobuf/ptypes/any"
+	api "github.com/osrg/gobgp/api"
+	"github.com/osrg/gobgp/internal/pkg/apiutil"
+	"github.com/osrg/gobgp/internal/pkg/config"
+	"github.com/osrg/gobgp/internal/pkg/table"
+	"github.com/osrg/gobgp/pkg/packet/bgp"
+)
+
+func marshalRouteTargets(l []string) ([]*any.Any, error) {
+	rtList := make([]*any.Any, 0, len(l))
+	for _, rtString := range l {
+		rt, err := bgp.ParseRouteTarget(rtString)
+		if err != nil {
+			return nil, err
+		}
+		rtList = append(rtList, apiutil.MarshalRT(rt))
+	}
+	return rtList, nil
+}
+
+func WatchConfig(configFile, configType string, apiServer *Server, isGracefulRestart bool) {
+	ch := make(chan *config.BgpConfigSet)
+	go config.ReadConfigfileServe(configFile, configType, ch)
+	configLoop := func() {
+		var c *config.BgpConfigSet
+		for {
+			log.Debug("Loading new configuration (new code)")
```

Comment:
I probably don't need to keep this Debug line. I should either remove it or make it an Info level log statement because it is kind of useful to get acknowledgement that the configuration was reloaded.